src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -5605,7 +5605,9 @@ public static StringAppendNode create() {
             final Rope left = libString.getRope(string);
```

Comment:
This `Pair<Rope, RubyEncoding>` above seems weirdly inconsistent with `RopeWithEncoding`, would be good to use `RopeWithEncoding` instead.
Unless somehow the returned Rope doesn't have the returned RubyEncoding?
Seems it's fine from a quick look.
cc @bjfish

Reply:
It must be correct to use RopeWithEncoding as the return value of executeStringAppend, because that's passed to methods which assert the encodings matches already anyway.

Reply:
Let's leave this as a cleanup for @bjfish since he knows this best.

Reply:
@eregon I think `RopeWithEncoding` would work the same here since `RopeWithEncoding` is pretty much a `Pair` itself.

> Unless somehow the returned Rope doesn't have the returned RubyEncoding?

`RopeWithEncoding` does have the assertion `assert rope.encoding == encoding.jcoding;` so we'll know if this scenario happens in testing.

Reply:
Okay. I'll leave this one and the previous comment about moving the `CheckStringEncodingCode` into a `@Cached` argument.